### ISSUE TITLE
AfxHook Error 13 - DLL can't be loaded (works normally this only happens when i use Reshade32.dll

### ISSUE THREAD

**Issue by Stra1f** [ripieces/advancedfx#904]
HLAE works if i don't use Reshade, but as soon as i add the reshade32.dll it gives me the error when trying to custom load. 
![asdasd121253235](https://github.com/advancedfx/advancedfx/assets/61309820/2f4199b4-bfdf-4550-b7cc-0ba08c0f7a29)

Here is my custom loader: 
![sd23534634](https://github.com/advancedfx/advancedfx/assets/61309820/2b06e379-eeb7-4c78-97ed-ee1c47790732)

i also tried adding the d3d9.dll as someone said that could work but no changes happens.

**Comment by dtugend** [comment]
CS2 is 64 bit, neither AfxHookSource nor ReShade32 will work with it.
You will need AfxHookSource2.dll (see Manual for the few commands it supports https://github.com/advancedfx/advancedfx/wiki/AfxHookSource2) in x64 sub-folder of HLAE and you will need ReShade64.dll.

The ReShade64 dll you can extract with https://7-zip.org/ from the ReShade full with addons download.
You might still need to create a ReShade.ini file in the <strike>cs2</strike> cs2.exe folder folder for it to work (don't forget to show file extensions or use some tool like e.g. [NotePad++](https://notepad-plus-plus.org/) to save it, otherwise you might end up with ReShade.ini.txt text file instead of ReShade.ini)

**Comment by Stra1f** [comment]
> CS2 is 64 bit, neither AfxHookSource nor ReShade32 will work with it. You will need AfxHookSource2.dll (see Manual for the few commands it supports https://github.com/advancedfx/advancedfx/wiki/AfxHookSource2) in x64 sub-folder of HLAE and you will need ReShade64.dll.
> 
> The ReShade64 dll you can extract with https://7-zip.org/ from the ReShade full with addons download. You might still need to create a ReShade.ini file in the cs2 folder for it to work (don't forget to show file extensions or use some tool like e.g. [NotePad++](https://notepad-plus-plus.org/) to save it, otherwise you might end up with ReShade.ini.txt text file instead of ReShade.ini)

TYSM! just tried doing it without the INI got this error: 
![asdas12423454235](https://github.com/advancedfx/advancedfx/assets/61309820/7abbb184-b935-42fc-9f31-484fba4f6a8c)

trying with the ini now

**Comment by Stra1f** [comment]
I just checked and ReShade.ini already exists in my cs2 folder

**Comment by dtugend** [comment]
It has to be C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Global Offensive\game\bin\win64, the cs2.exe folder, not the cs2 folder.

**Comment by Stra1f** [comment]
> It has to be C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Global Offensive\game\bin\win64, the cs2.exe folder, not the cs2 folder

yeah it's already in there 
![asdasd122522](https://github.com/advancedfx/advancedfx/assets/61309820/f2cf68e8-7d71-4f1b-9cd5-b07aec34f7ca)

**Comment by dtugend** [comment]
this is a .txt, not a .ini

**Comment by Stra1f** [comment]
oh when i tried to add it with an ini it said it already existed let me try again

**Comment by dtugend** [comment]
[ReShade.zip](https://github.com/advancedfx/advancedfx/files/14182565/ReShade.zip)
Try this one.

**Comment by dtugend** [comment]
also remove -game csgo

**Comment by Stra1f** [comment]
same error :(

**Comment by dtugend** [comment]
What are your current settings?
Which graphic card vendor do you use? NVIDIA or AMD?

**Comment by Stra1f** [comment]
Nvidia Card

settings in game? i havent toutched HLAE settings

**Comment by dtugend** [comment]
![image](https://github.com/advancedfx/advancedfx/assets/3610128/65a462e5-bc9e-49da-9f28-291f3f7f92af)

> settings in game? i havent toutched HLAE settings

Ingame settigns don't matter much - You need to remove -game csgo as said

**Comment by Stra1f** [comment]
i did

**Comment by Stra1f** [comment]
![Uploading se1q24135235.PNG…]()

**Comment by dtugend** [comment]
Can you post your current custom loader settings?
Also which ReShade version did you use?
I used latest from reshade.me

**Comment by Stra1f** [comment]
![se1q24135235](https://github.com/advancedfx/advancedfx/assets/61309820/0ac7b1a8-1f5e-4cd7-9697-cf6a7c76af3a)

I used latest from ReShade.me and the one with plugin support

**Comment by Stra1f** [comment]
is there supposed to be a text reshade? when i remove it it finds its way back after launching cs 
![Uploading se1q24135235.PNG…]()

**Comment by Stra1f** [comment]
![se1q24135235](https://github.com/advancedfx/advancedfx/assets/61309820/89771f5e-07de-4354-a30f-f9b350a37a08)
is there supposed to be a text reshade? when i remove it it finds its way back after launching cs

**Comment by Stra1f** [comment]
is there supposed to be a text reshade? when i remove it it finds its way back after launching cs
![asdas124235235](https://github.com/advancedfx/advancedfx/assets/61309820/5c311162-6273-48f6-9020-79a6662f3b5a)

**Comment by dtugend** [comment]
Can you verify the game cache first and make sure you are not signed into any beta?

**Comment by Stra1f** [comment]
If this helps i also selected DirectX11 when downloading Reshade.

**Comment by Stra1f** [comment]
> Can you verify the game cache first and make sure you are not signed into any beta?

doing that rn

**Comment by Stra1f** [comment]
could try and install with vulkan instead of direct x 11 or do you not think that would change anything?

**Comment by Stra1f** [comment]
same error after verifying.

**Comment by dtugend** [comment]
On the official page reshade.me you don't need to select anything.
You can just download the exe and then use 7-zip.org as said to get the right DLL.

If you did install reshade into cs2.exe folder, please make sure to revert that (remove d3d11.dll and rehade.ini and make a new empty one).

You can also check the ReShade.log file it creates to see what errors are there.

I never tested with Volkan

**Comment by Stra1f** [comment]
> If you did install reshade into cs2.exe folder, please make sure to revert that (remove d3d11.dll and rehade.ini and make a new empty one).

So just so i get this right, i should not install reshade? And instead just grab the ReShade64.dll and continue with HLAE

**Comment by dtugend** [comment]
Well it's easier to install it if you want the shaders and textures to be honest, but then you need to revert the d3d11.dll etc.

**Comment by Stra1f** [comment]
Won't work without installing, trying that now

**Comment by dtugend** [comment]
I mean d3d11.dll only in the cs2.exe folder not elsewhere

**Comment by Stra1f** [comment]
hmm, even without installed or with it never creates a d3d11.dll

**Comment by dtugend** [comment]
ok, can you check the reshade.log then if it's still same error as before?

**Comment by Stra1f** [comment]
![Uploading 234234321123.PNG…]()

**Comment by Stra1f** [comment]
![234234321123](https://github.com/advancedfx/advancedfx/assets/61309820/2a9dd6f9-f9bf-41c1-8278-3cc58d6039e9)

**Comment by dtugend** [comment]
just delete the dxgi.dll in CS2.exe folder.
(It's from when you installed reshade)

**Comment by Stra1f** [comment]
TYSM! worked. 
never expected someone to reply so fast, looked at some of the older issues and a lot of them were unanswered xD
TY! <3

**Comment by dtugend** [comment]
Do the shaders etc. work?
If you have problems with the depth buffer disable FidelityFX Super Resolution in video advanced settings and MSAA.
If you have other questions people can usually help in the Discord linked on the support page.

**Comment by Stra1f** [comment]
I have one last issue witch no worries if you don't know about this. But stuff like Depth Of Field does not work, if i try and enable the display depth it's just 2 different colors split on screen rather than an actual DOF map. I think i saw something in a tutorial about a fix with the def buffer. Ill try to find that again while you can let me know if you know how to fix that.

**Comment by dtugend** [comment]
> If you have problems with the depth buffer disable FidelityFX Super Resolution in video advanced settings and [disable] MSAA.

**Comment by Stra1f** [comment]
yeah i already disabled that. I think i fixed it!

**Comment by Stra1f** [comment]
yeah this looks like a proper DOF map 
![se1242346346](https://github.com/advancedfx/advancedfx/assets/61309820/855dd5a0-b68e-425b-be9d-93a542eea8f3)

**Comment by dtugend** [comment]
Probably need to adjust min/max values a bit etc

**Comment by dtugend** [comment]
![image](https://github.com/advancedfx/advancedfx/assets/3610128/126bcf85-fbe8-409f-a830-50987fd85667)

**Comment by Stra1f** [comment]
should i change the values on the DOF effect im using? (MartysDOF) or is there a general DOF setting place?

**Comment by Stra1f** [comment]
some of the dofs just work. But the best one (someone said) (martyDOF) it blurs the gun, and no settings seem to change that. 

https://github.com/advancedfx/advancedfx/assets/61309820/f8028577-1e5a-4aa9-8e56-919714c1103a

**Comment by dtugend** [comment]
The DOF effect has a preview, find good values there, then turn it off and put them into global pre-processor options
ALso turn of "Reversed Depth" there, which is on my default.

**Comment by dtugend** [comment]
You want to set DOF_NEARBLURCURVE to 10 or 100

**Comment by Stra1f** [comment]
Gotcha! also found a way for the dof to ignore the weapon viewmodel so that's ignored now. Now only thing that confuses me. So every single tutorial online about ReShade DOF on csgo and cs2 says to write the command "r_DrawVgui" but apparently this was removed a few months back. Is it not longer needed? or am i fucked for not being able to turn that on. It works but right now it looks like the autofocus is terrible (not sure if that's what VGUI was supposed to fix) but the autofocus changes is happening way to often and disturbs the gameplay

**Comment by Stra1f** [comment]
would love to just have it kind of static without autofocusing

**Comment by Stra1f** [comment]
i just disabled AutoFocus and setup a value that looked good that works way better. Still confused about vgui and if that just doesn't matter anymore. Cuz this looks good to me

**Comment by dtugend** [comment]
Dunno, haven't worked much with CS2 ReShade yet tbh.
`mirv_cvar_unhide_all; sv_cheats 1; r_drawpanorama false`

**Comment by Stra1f** [comment]
have you come across an animation issue where the gun fx. jiggles about. Pretty much ruins the demo completely

**Comment by Stra1f** [comment]
happens when ppl shoot too, the weapon just offsets from the hand or vice versa 
![sd3253456346](https://github.com/advancedfx/advancedfx/assets/61309820/ba060039-6297-4f31-864e-9714788c00f6)

**Comment by Dechno1337** [comment]
> have you come across an animation issue where the gun fx. jiggles about. Pretty much ruins the demo completely

That's something Valve needs to fix.

**Comment by SandmX** [comment]
Compared to AfxHookSource, -2 lacks many useful features such as "mirv-loadlibrary"
